### PR TITLE
APIS-6721 Upgrade mongo lib to 1.7.0

### PR DESCRIPTION
**Pull request by petekirby-ee**

### CODE REVIEW

File: app/uk/gov/hmrc/apiplatform/modules/common/services/InstantSyntax.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/*
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/apiplatform/modules/common/services/InstantSyntax.scala]: It should be the one in the lib now (common domain - might mean approving/fixing John's PR) so this doesn't proliferate

File: app/uk/gov/hmrc/apiplatformjobs/scheduled/DeleteUnverifiedDevelopersJob.scala [hunk 2]
```diff
@@ -63,7 +64,7 @@ class DeleteUnverifiedDevelopersJob @Inject() (
     logger.info("Starting DeleteUnverifiedDevelopersJob")
 
     (for {
-      developerDetails <- developerConnector.fetchUnverifiedDevelopers(now().minusDays(createdBeforeInDays), jobConfig.limit)
+      developerDetails <- developerConnector.fetchUnverifiedDevelopers(instant().minus(createdBeforeInDays, ChronoUnit.DAYS), jobConfig.limit)
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/apiplatformjobs/scheduled/DeleteUnverifiedDevelopersJob.scala]: maybe use Duration.ofDays(createdBeforeInDays) - not sure that this is the "new" standard form

File: app/uk/gov/hmrc/apiplatformjobs/scheduled/TimedJob.scala [hunk 3]
```diff
@@ -50,10 +50,10 @@ abstract class TimedJob @Inject() (override val name: String, configuration: Con
   override def interval: FiniteDuration = jobConfig.executionInterval.interval
 
   def calculateInitialDelay(timeOfFirstRun: LocalTime): FiniteDuration = {
-    val currentDateTime        = LocalDateTime.now(clock)
-    val timeToday              = LocalDateTime.of(LocalDate.now(clock), timeOfFirstRun)
-    val nextInstanceOfTime     = if (timeToday.isBefore(currentDateTime)) timeToday.plusDays(1) else timeToday
-    val millisecondsToFirstRun = nextInstanceOfTime.toInstant(UTC).toEpochMilli - currentDateTime.toInstant(UTC).toEpochMilli
+    val currentDateTime        = Instant.now(clock)
+    val timeToday              = LocalDateTime.of(LocalDate.now(clock), timeOfFirstRun).toInstant(ZoneOffset.UTC)
+    val nextInstanceOfTime     = if (timeToday.isBefore(currentDateTime)) timeToday.plus(1, ChronoUnit.DAYS) else timeToday
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/apiplatformjobs/scheduled/TimedJob.scala]: Ditto

File: app/uk/gov/hmrc/apiplatformjobs/scheduled/UpdateUnusedApplicationRecordsJob.scala [hunk 4]
```diff
@@ -42,8 +42,8 @@ abstract class UpdateUnusedApplicationRecordsJob(
 
   /** The date we should use to find applications that have not been used since. This should be far enough in advance that all required notifications can be sent out.
     */
-  def notificationCutoffDate(): LocalDateTime =
-    LocalDateTime
+  def notificationCutoffDate(): Instant =
+    Instant
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/apiplatformjobs/scheduled/UpdateUnusedApplicationRecordsJob.scala]: Transform this to instant() and use ClockNow maybe to ensure truncation.

File: app/uk/gov/hmrc/apiplatformjobs/scheduled/UpdateUnusedApplicationRecordsJob.scala [hunk 5]
```diff
@@ -105,7 +105,7 @@ abstract class UpdateUnusedApplicationRecordsJob(
   def unusedApplicationRecord(applicationUsageDetails: ApplicationUsageDetails, verifiedAdministratorDetails: Map[LaxEmailAddress, Administrator]): UnusedApplication = {
     val verifiedApplicationAdministrators =
       applicationUsageDetails.administrators.intersect(verifiedAdministratorDetails.keySet).flatMap(verifiedAdministratorDetails.get)
-    val lastInteractionDate               = applicationUsageDetails.lastAccessDate.getOrElse(applicationUsageDetails.creationDate).toLocalDate
+    val lastInteractionDate               = LocalDate.ofInstant(applicationUsageDetails.lastAccessDate.getOrElse(applicationUsageDetails.creationDate), ZoneOffset.UTC)
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/apiplatformjobs/scheduled/UpdateUnusedApplicationRecordsJob.scala]: Could use the common lib syntaxes to go asLocalDateTime or asInstant

File: test/uk/gov/hmrc/apiplatformjobs/connectors/ApplicationCommandConnectorSpec.scala [hunk 6]
```diff
@@ -37,11 +37,11 @@ import uk.gov.hmrc.apiplatform.modules.common.domain.models.{Actors, Application
 import uk.gov.hmrc.apiplatformjobs.models.HasSucceeded
 import uk.gov.hmrc.apiplatformjobs.util.{AsyncHmrcSpec, UrlEncoding}
 
-class ApplicationCommandConnectorSpec extends AsyncHmrcSpec with RepsonseUtils with GuiceOneAppPerSuite with WiremockSugar with UrlEncoding {
+class ApplicationCommandConnectorSpec extends AsyncHmrcSpec with ResponseUtils with GuiceOneAppPerSuite with WiremockSugar with UrlEncoding {
 
   val appId        = ApplicationId.random
   val actor        = Actors.ScheduledJob("TestMe")
-  val timestamp    = LocalDateTime.now()
+  val timestamp    = Instant.now()
```
Comment (AndySpaven) [inline test/uk/gov/hmrc/apiplatformjobs/connectors/ApplicationCommandConnectorSpec.scala]: Time to bring in fixedclock maybe ?

File: test/uk/gov/hmrc/apiplatformjobs/connectors/ApplicationCommandConnectorSpec.scala [hunk 7]
```diff
@@ -73,7 +73,7 @@ class ApplicationCommandConnectorSpec extends AsyncHmrcSpec with RepsonseUtils w
       val developerCollaborator = Collaborators.Developer(UserId.random, "dev@example.com".toLaxEmail)
       val jsonText              =
         s"""{"command":{"actor":{"email":"${anAdminEmail.text}","actorType":"COLLABORATOR"},"collaborator":{"userId":"${developerCollaborator.userId.value}","emailAddress":"dev@example.com","role":"DEVELOPER"},"timestamp":"2020-01-01T12:00:00Z","updateType":"removeCollaborator"},"verifiedCollaboratorsToNotify":["admin@example.com"]}"""
-      val timestamp             = LocalDateTime.of(2020, 1, 1, 12, 0, 0)
+      val timestamp             = LocalDateTime.of(2020, 1, 1, 12, 0, 0).toInstant(ZoneOffset.UTC)
```
Comment (AndySpaven) [inline test/uk/gov/hmrc/apiplatformjobs/connectors/ApplicationCommandConnectorSpec.scala]: using syntax..

File: test/uk/gov/hmrc/apiplatformjobs/connectors/ApplicationCommandConnectorSpec.scala [hunk 8]
```diff
@@ -90,7 +90,7 @@ class ApplicationCommandConnectorSpec extends AsyncHmrcSpec with RepsonseUtils w
       val developerCollaborator = Collaborators.Developer(UserId.random, "dev@example.com".toLaxEmail)
       val jsonText              =
         s"""{"command":{"actor":{"jobId":"BOB","actorType":"SCHEDULED_JOB"},"collaborator":{"userId":"${developerCollaborator.userId.value}","emailAddress":"dev@example.com","role":"DEVELOPER"},"timestamp":"2020-01-01T12:00:00Z","updateType":"removeCollaborator"},"verifiedCollaboratorsToNotify":[]}"""
-      val timestamp             = LocalDateTime.of(2020, 1, 1, 12, 0, 0)
+      val timestamp             = LocalDateTime.of(2020, 1, 1, 12, 0, 0).toInstant(ZoneOffset.UTC)
```
Comment (AndySpaven) [inline test/uk/gov/hmrc/apiplatformjobs/connectors/ApplicationCommandConnectorSpec.scala]: using syntax..

File: test/uk/gov/hmrc/apiplatformjobs/connectors/ThirdPartyApplicationConnectorSpec.scala [hunk 9]
```diff
@@ -126,26 +126,27 @@ class ThirdPartyApplicationConnectorSpec extends AsyncHmrcSpec with RepsonseUtil
     val allowAutoDelete                  = true
 
     "return application details as ApplicationUsageDetails objects" in new Setup {
-      val lastUseDate        = LocalDateTime.now.minusMonths(12)
-      val dateString: String = dateFormatter.format(lastUseDate)
+      val lastUseLDT         = LocalDateTime.now.minusMonths(12)
+      val lastUseDate        = lastUseLDT.toInstant(ZoneOffset.UTC)
```
Comment (AndySpaven) [inline test/uk/gov/hmrc/apiplatformjobs/connectors/ThirdPartyApplicationConnectorSpec.scala]: using syntax.. Could just become `now.minusMonths(12).asInstant()`

**Review by AndySpaven — CHANGES_REQUESTED** [review]
So lots of stuff that could move to ClockNow/FixedClock and the syntaxes that JG added to common lib.

File: app/uk/gov/hmrc/apiplatformjobs/scheduled/DeleteUnverifiedDevelopersJob.scala (hunk 2 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/apiplatformjobs/scheduled/DeleteUnverifiedDevelopersJob.scala]: Fixed

File: app/uk/gov/hmrc/apiplatformjobs/scheduled/TimedJob.scala (hunk 3 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/apiplatformjobs/scheduled/TimedJob.scala]: Fixed

File: test/uk/gov/hmrc/apiplatformjobs/connectors/ThirdPartyApplicationConnectorSpec.scala (hunk 9 above)
Comment (petekirby-ee) [reply test/uk/gov/hmrc/apiplatformjobs/connectors/ThirdPartyApplicationConnectorSpec.scala]: Fixed

File: test/uk/gov/hmrc/apiplatformjobs/connectors/ApplicationCommandConnectorSpec.scala (hunk 6 above)
Comment (petekirby-ee) [reply test/uk/gov/hmrc/apiplatformjobs/connectors/ApplicationCommandConnectorSpec.scala]: Done

File: app/uk/gov/hmrc/apiplatformjobs/scheduled/UpdateUnusedApplicationRecordsJob.scala (hunk 4 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/apiplatformjobs/scheduled/UpdateUnusedApplicationRecordsJob.scala]: Done

File: app/uk/gov/hmrc/apiplatformjobs/scheduled/UpdateUnusedApplicationRecordsJob.scala (hunk 5 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/apiplatformjobs/scheduled/UpdateUnusedApplicationRecordsJob.scala]: Done

File: test/uk/gov/hmrc/apiplatformjobs/connectors/ApplicationCommandConnectorSpec.scala (hunk 7 above)
Comment (petekirby-ee) [reply test/uk/gov/hmrc/apiplatformjobs/connectors/ApplicationCommandConnectorSpec.scala]: Done

File: test/uk/gov/hmrc/apiplatformjobs/connectors/ApplicationCommandConnectorSpec.scala (hunk 8 above)
Comment (petekirby-ee) [reply test/uk/gov/hmrc/apiplatformjobs/connectors/ApplicationCommandConnectorSpec.scala]: Done

File: app/uk/gov/hmrc/apiplatform/modules/common/services/InstantSyntax.scala (hunk 1 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/apiplatform/modules/common/services/InstantSyntax.scala]: Done

**Review by AndySpaven — APPROVED** [review]

**Pull request merged**